File: examples/JRubyExample/print_Cobol.rb
################################################################
## This is a simple example of accessing cb2xml from ruby
## It simply prints the content of the Xml
## Because we are reading the Cobol copybook (rather than the Xml)
## we need the cb2xml.jar and cb2xml_Jaxb.jar
##
## Author: Bruce Martin
## License: Any, e.g. LGPL (any), Apache, Creative Commons
################################################################
require '../../lib/cb2xml.jar'
require '../../lib/cb2xml_Jaxb.jar'

java_import 'net.sf.cb2xml.parse.CobolParser'
  
   def printXml indent, items

       items.each do |item|
           puts "#{indent} #{item.getLevel()} #{item.getName()}\t#{item.getPosition()}\t#{item.getStorageLength()}\t#{item.getPicture()}\t#{item.getUsage()}"   
           printXml "#{indent}   ", item.getItem()   
       end
   end

   copybook = CobolParser.new().parseCobol("cbl2xml_Test110.cbl")
   
   printXml "    ", copybook.getItem()
   
   x = gets

File: examples/JythonExample/print_Cobol.py
##############################################################
#
#  Purpose: Illustrate processing a Cobol Copybook in jython.
#           It prints the  Cobol Copybook content to the terminal
#           Because is processing Cobol, this program needs both the
#             * cb2xml.jar (Cobol Processing
#             * cb2xml_Jaxb (Xml JAXB stuff)
#   Author: Bruce Martin
#  License: any e.g. LGPL (any), Apache, Creative Commons
#
##############################################################

import sys

sys.path.append("../../lib/cb2xml.jar")
sys.path.append("../../lib/cb2xml_Jaxb.jar")

from net.sf.cb2xml.parse import CobolParser

## return the value or '' if null. 
def fix(v):
    r="\t"
    if v != None:
        r="\t" + v
    return r

def fixBoolean(a, v):
    r=''
    if v != None:
        r=a + "=true" 
    return r

 
##########################################################################
# Purpose: Print one item and print any children
##########################################################################
def printItem(indent, item):
    n = indent + item.getLevel() + " " + item.getName() + "                                                                                      "
    n = n[:50]
    print n, '\t', item.getPosition(), '\t', item.getStorageLength(), fix(item.getPicture()), fix(item.getUsage()), fixBoolean("numeric",item.isNumeric()), fixBoolean("signed", item.isSigned())
    children = item.getItem()
    for child in children:
	printItem(indent + "   ", child)

#########################################################################

copybook = CobolParser().parseCobol("cbl2xml_Test110.cbl")

print ">> ", copybook.getFilename()

children = copybook.getItem()
for child in children:
    printItem("    ", child)
    

